fvtest/compilertriltest/ArithmeticTest.cpp

```diff
@@ -89,6 +89,22 @@ int64_t lrem(int64_t l, int64_t r) {
     return l%r;
 }
 
+int16_t sadd(int16_t l, int16_t r) {
+    return l+r;
+}
+
+int16_t ssub(int16_t l, int16_t r) {
+    return l-r;
+}
+
+int8_t badd(int8_t l, int8_t r) {
+    return l+r;
+}
+
+int8_t bsub(int8_t l, int8_t r) {
+    return l-r;
+}
```

Comment:
Is it possible to template these functions? The only difference between them seems to be the type. We can save a lot of duplication and make things much more readable.

Reply:
At one point, these functions were implemented as a template but one of our supported build compilers (MSVC 2010, I believe) failed to properly deduce types. So, we had to use separate implementations for the different types. Maybe this is something worth revisiting? I don't believe we use MSVC 2010 anymore for any of our CI builds, but I'm not sure if we still officially support it.

Reply:
> I don't believe we use MSVC 2010 anymore for any of our CI builds, but I'm not sure if we still officially support it.

I would wager that what we use for our builds is our minimal supported level, otherwise we're bound to break support if we're not testing against MSVC 2010. But yes, perhaps we can give this another quick try.

Reply:
So is the conclusion that we can't accept this as is and we are asking @midronij to template the functions?

Reply:
I think templating the function is beyond the scope of this PR because doing so will require extra investigation to verify it will work correctly on all our supported build compilers. Instead, I will open an issue to do the conversion work, which @Yuehan-Lin has agreed to look into.

Reply:
Just opened #4258.

Reply:
Excellent. I'm satisfied with the resolution here. We seem to have a path forward and this PR can be accepted as is.